File: src/io/dotenv.ts
// Minimal dotenv parser for `menv import` — parse only (generation/serialization
// is Plan 3). Values are single-line (the v2 spec keeps v1's restriction); no
// escape-sequence processing beyond removing one layer of surrounding quotes.
export interface DotenvEntry {
  key: string;
  value: string;
}

export function parseDotenv(text: string): DotenvEntry[] {
  const out: DotenvEntry[] = [];
  for (const rawLine of text.split("\n")) {
    const line = rawLine.trim();
    if (line === "" || line.startsWith("#")) continue;
    const body = line.startsWith("export ") ? line.slice("export ".length).trim() : line;
    const eq = body.indexOf("=");
    if (eq === -1) continue;
    const key = body.slice(0, eq).trim();
    let value = body.slice(eq + 1).trim();
    const quoted =
      value.length >= 2 &&
      ((value.startsWith('"') && value.endsWith('"')) || (value.startsWith("'") && value.endsWith("'")));
    if (quoted) {
      value = value.slice(1, -1);
    } else {
      const hash = value.indexOf(" #");
      if (hash !== -1) value = value.slice(0, hash).trimEnd();
    }
    out.push({ key, value });
  }
  return out;
}

File: go/internal/io/dotenv.go
package io

import "strings"

// DotenvEntry is a key=value pair parsed from a .env file.
type DotenvEntry struct {
	Key   string
	Value string
}

// ParseDotenv parses dotenv-format text: strips comments, blank lines, and
// optional surrounding quotes. No escape-sequence processing (v2 keeps single-
// line values only). The `export ` prefix is accepted and stripped.
func ParseDotenv(text string) []DotenvEntry {
	var out []DotenvEntry
	for _, rawLine := range strings.Split(text, "\n") {
		line := strings.TrimSpace(rawLine)
		if line == "" || strings.HasPrefix(line, "#") {
			continue
		}
		body := line
		if strings.HasPrefix(body, "export ") {
			body = strings.TrimSpace(body[len("export "):])
		}
		eq := strings.IndexByte(body, '=')
		if eq == -1 {
			continue
		}
		key := strings.TrimSpace(body[:eq])
		value := strings.TrimSpace(body[eq+1:])
		if len(value) >= 2 &&
			((value[0] == '"' && value[len(value)-1] == '"') ||
				(value[0] == '\'' && value[len(value)-1] == '\'')) {
			value = value[1 : len(value)-1]
		} else {
			// Inline comment: strip everything from " #" onward.
			if idx := strings.Index(value, " #"); idx != -1 {
				value = strings.TrimRight(value[:idx], " \t")
			}
		}
		out = append(out, DotenvEntry{Key: key, Value: value})
	}
	return out
}
